server-base/src/main/java/org/apache/kylin/rest/service/QueryService.java

```diff
@@ -509,13 +523,16 @@ private SQLResponse queryAndUpdateCache(SQLRequest sqlRequest, boolean queryCach
                     String.valueOf(sqlResponse.getIsException()), String.valueOf(sqlResponse.getDuration()),
                     String.valueOf(sqlResponse.getTotalScanCount()));
             if (checkCondition(queryCacheEnabled, "query cache is disabled") //
+                    && checkCondition(!Strings.isNullOrEmpty(sqlResponse.getCube()),
+                            "query does not hit cube nor hybrid") //
                     && checkCondition(!sqlResponse.getIsException(), "query has exception") //
                     && checkCondition(
                             !(sqlResponse.isPushDown()
                                     && (isSelect == false || kylinConfig.isPushdownQueryCacheEnabled() == false)),
                             "query is executed with pushdown, but it is non-select, or the cache for pushdown is disabled") //
                     && checkCondition(
-                            sqlResponse.getDuration() > durationThreshold
+                            cacheManager.getCache(QUERY_CACHE) instanceof MemcachedCacheManager.MemCachedCacheAdaptor
```

Comment:
too lightweight still cached in memcached? will the overhead brought by memcached slow down the query?

Reply:
Since memcached is a distributed cache and it's easy to scale out, we don't worry about the capacity issue of memcached. 
Compared to go through hbase, the overhead to call memcached is very little. What's more, call memcached instead of hbase can reduce lots of rpc to hbase. When hbase is not stable, this can also improve the HA in some degree.

Reply:
memcached和default逻辑不一样，这样有些太复杂了；对于lightweight query cache的处理有配置管理了，应该没必要通过memcached来屏蔽。

建议把这个判断去掉，就是说使用memcached时依然允许不缓存lightweight query；如果像你说的想减轻hbase负担，可以设置durationThreshold=0和scanCountThreshold=0等参数，作为性能优化配置的一部分。

Reply:
还是觉得应该对不同的cachemanager做不同的处理。像durationThreshold等参数更适合local的cachemanager，而不是MemcachedCacheManager。这里如果把这个逻辑去掉只用durationThreshold等参数的话，就不能更好地利用RemoteLocalFailOverCacheManager的特性了。有了这个参数之后，当memcached有问题用本地的cachemanager的时候，durationThreshold等参数就有效果了。

Reply:
不过针对一些探测查询，如select 1，确实没必要cache。